### PR TITLE
Add basic figshare API interaction

### PR DESCRIPTION
**Pull request by vasudevanv**
Added some code with a figshare class and methods to create and update articles. Figuring out how to handle OAuth2.

### CODE REVIEW

**Comment by vasudevanv** [comment]
Need to address
* Add functionality to download specific versions of articles with version=None referring to the latest version.
* Download article without tokens
* Add some simple function wrappers

File: figshare/figshare.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import sys
+import os
+import hashlib
+import json
+import requests
+
+from requests.exceptions import HTTPError
+
+
+def issue_request(method, url, headers, data=None, binary=False):
```
Comment (dhimmel) [inline figshare/figshare.py:11]: Let's add [docstrings](http://docs.python-guide.org/en/latest/writing/documentation/) describing what the function does and what the parameters are.

File: figshare/figshare.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        if "error" not in response.keys():
+            article_id = int(response["location"].split("/")[-1])
+        else:
+            article_id = None
+        return article_id
+
+    def update_article(self, article_id, **kwargs):
+        """ Updates an article with a given article_id. """
+        allowed = {'title', 'description', 'defined_type', 'tags', 'categories'}
+        valid_keys = set(kwargs.keys()).intersection(set(allowed))
```
Comment (dhimmel) [inline figshare/figshare.py]: `set(kwargs.keys())` is the same as `set(kwargs)`. `allowed` is already a set.

File: figshare/figshare.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        body = {}
+
+        for key in valid_keys:
+            body[key] = kwargs[key]
+
+        url = self.endpoint('/account/articles/{0}'.format(str(article_id)))
```
Comment (dhimmel) [inline figshare/figshare.py]: `str` not needed, format will automatically convert into to string.

File: figshare/figshare.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                'defined_type': defined_type,
+                'categories': categories,
+                'tags': tags}
+
+        url = self.endpoint("/account/articles")
+        headers = self.get_headers(self.token)
+        response = self.issue_request('POST', url, headers=headers, data=data)
+        response = json.loads(response.body_string())
+
+        if "error" not in response.keys():
```
Comment (dhimmel) [inline figshare/figshare.py]: `.keys()` not needed here.

File: figshare/figshare.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import sys
+import os
+import hashlib
+import json
```
Comment (dhimmel) [inline figshare/figshare.py:4]: Newline after last builtin import.

**Review by dhimmel — COMMENTED** [review]
Some minor comments. Main thing is documentation of functions.

**Comment by vasudevanv** [comment]
@dhimmel Updated the code based on your comments. Also added a setup script. Going to add more docstrings before merging.

**Review by dhimmel — APPROVED** [review]
@vasudevanv looks great.

I will merge this now. Excited to try it out.

**Pull request merged**